### PR TITLE
KL_Div+Entropy changes , PL / ROI config flag for DINO

### PR DESCRIPTION
**Pull request by adv010**

### CODE REVIEW

File: pcdet/ssod/contrastive.py [hunk 1]
```diff
@@ -166,11 +166,21 @@ def forward(self, batch_dict_wa_lbl, batch_dict_wa_ulb, batch_dict_sa_lbl, batch
...
-                sa_roi_labels = pls[..., -1].long().view(-1) - 1  # 0-based
-                cls_weights = self.cls_weights.to(sa_roi_labels.device)[sa_roi_labels].view(-1)
-                keep_mask = torch.logical_not(torch.eq(wa_rois, 0).all(dim=-1)).view(-1).float()
+                if self.cfgs.MODEL.DINO_HEAD.INPUT=='ROI':
+                    sa_rois = batch_dict_sa_ulb['rois'] # passing sa_rois without clone leads to in-place modification
+                    sa_roi_labels = batch_dict_sa_ulb['roi_labels'].long().view(-1) - 1
+                    keep_mask = self.mask_dense_rois(batch_dict_sa_ulb['points'], sa_rois).view(-1).float()
+                    cls_weights = self.cls_weights.to(sa_roi_labels.device)[sa_roi_labels].view(-1)
+                    wa_rois = transform_aug(sa_rois, batch_dict_sa_ulb, batch_dict_wa_ulb)
+                    masked_rois_ratio = keep_mask.sum() / sa_rois.shape[0]
```
Comment (fnozarian) [inline pcdet/ssod/contrastive.py:175]: Rename masked_rois_ratio to keep_mask_ratio or keep_ratio. Mask is usually referred to the ones ignored. Also, move it outside if/else to let see the ratio for both cases.

**Review by fnozarian — APPROVED** [review]
I'll apply the changes described [here](https://github.com/fnozarian/OpenPCDet/pull/51/files#r1929030942).
